org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessParameterPanel.java

```diff
@@ -654,6 +654,14 @@ public boolean saveParameters() {
 			//Save only parameters which are set
 			if((result == null) && (result2 == null))
 				continue;
+			if(result instanceof String) {
+				if((result2 == null) && (Util.isEmpty(String.valueOf(result))))
+					continue;
+				else if((result2 instanceof String)
+						&& (Util.isEmpty(String.valueOf(result)) 
+						&& Util.isEmpty(String.valueOf(result2))))
+					continue;
```

Comment:
this whole block can be simplify to if (Util.isEmpty((String)result) && Util.isEmpty((String)result2)) continue.

Reply:
`if(result instanceof String) {
    if (Util.isEmpty((String)result) && Util.isEmpty((String)result2))
        continue`

Reply:
sorry, my mistake, the above is not correct since result2 can be null. should be `Util.isEmpty((String)result) && (result2 == null || Util.isEmpty((String)result2))` instead.